conans/client/command.py

```diff
@@ -1358,21 +1358,29 @@ def get(self, *args):
                                  'conanfile if only a reference is specified and a conaninfo.txt '
                                  'file contents if the package is also specified',
                             default=None, nargs="?")
-        parser.add_argument("-p", "--package", default=None, help='Package ID',
+        parser.add_argument("-p", "--package", default=None, help=argparse.SUPPRESS,
                             action=OnceArgument)
         parser.add_argument("-r", "--remote", action=OnceArgument,
                             help='Get from this specific remote')
         parser.add_argument("-raw", "--raw", action='store_true', default=False,
                             help='Do not decorate the text')
         args = parser.parse_args(*args)
 
-        ret, path = self._conan.get_path(args.reference, args.package, args.path, args.remote)
-        if isinstance(ret, list):
-            self._outputer.print_dir_list(ret, path, args.raw)
-        else:
-            self._outputer.print_file_contents(ret, path, args.raw)
+        try:
+            pref = PackageReference.loads(args.reference)
+            reference = pref.ref.full_repr()
+            package_id = pref.id
+        except ConanException:
+            reference = args.reference
+            package_id = args.package
+        finally:
```

Comment:
This finally is very weird, it means it will always get executed, even if there is some random Exception in the try. Please remove the finally.